zipline/algorithm.py

```diff
@@ -605,17 +651,32 @@ def record(self, *args, **kwargs):
     def symbol(self, symbol_str):
         """
         Default symbol lookup for any source that directly maps the
-        symbol to the identifier (e.g. yahoo finance).
+        symbol to the Asset (e.g. yahoo finance).
         """
-        return symbol_str
+        asset, _ = trading.environment.asset_finder.lookup_generic(
+            asset_convertible_or_iterable=symbol_str,
+            as_of_date=self.datetime,
+            )
+        return asset
 
     @api_method
     def symbols(self, *args):
         """
         Default symbols lookup for any source that directly maps the
-        symbol to the identifier (e.g. yahoo finance).
+        symbol to the Asset (e.g. yahoo finance).
         """
-        return args
+        result = []
```

Comment:
This definitely works. And sometimes this form makes it easier to step through etc.
But this kind of transform lends itself semantically to being a comprehension, e.g. `return [self.symbol(identifier) for identifier in args]`